libfeed/feed-store-provider.c

```diff
@@ -119,20 +121,31 @@ call_dbus_proxy_and_construct_from_model (GDBusProxy           *proxy,
 }
 
 static gchar *
-remove_uri_prefix (const gchar *uri, const gchar *prefix)
+strip_leading_slashes (const gchar *str)
+{
+  const gchar *strp = str;
+
+  while (*strp == '/')
+    ++strp;
+
+  return g_strdup (strp);
+}
+
+static gchar *
+remove_uri_prefix (const gchar *uri)
 {
-  g_autoptr(GFile) uri_parent = g_file_new_for_uri (prefix);
-  g_autoptr(GFile) uri_file = g_file_new_for_uri (uri);
```

Comment:
You can actually use GIO, but you have to be a bit more careful to use the right amount of slashes :-)
```
gjs> const Gio = imports.gi.Gio;
gjs> var parent = Gio.File.new_for_uri('ekn:///');
gjs> var thumbnail = Gio.File.new_for_uri('ekn://asdasdasd')
gjs> var relpath = parent.get_relative_path(thumbnail);
gjs> relpath
null
gjs> thumbnail = Gio.File.new_for_uri('ekn:///asdasdasd')
[object instance proxy GType:GDaemonFile jsobj@0x7f622869eb50 native@0x7f61e8011580]
gjs> relpath = parent.get_relative_path(thumbnail);
"asdasdasd"
gjs> 
```

Note how I had to use `ekn:///` as parent (triple-slash) and `ekn:///path` (again, triple slash) as the children.
Can you use that here?

Reply:
So, that does not work if you have an `ekn://` path with double-slashes:

```
gjs> Gio.File.new_for_uri('ekn:///').get_relative_path(Gio.File.new_for_uri('ekn://foo'))
null
gjs> Gio.File.new_for_uri('ekn:///').get_relative_path(Gio.File.new_for_uri('ekn:///foo'))
"foo"
```

To be honest, I'd rather just be robust against the edge case here of some app being built using double slashes instead of triple slashes for thumbnail URIs. Seems like parsing the URI properly is the way to go.

When GUri lands in glib upstream we can drop the dependency.

Reply:
Alright, let's go with your recommendation.